bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java

```diff
@@ -0,0 +1,313 @@
+
+    public TelldusLocalBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    private ScheduledFuture<?> pollingJob;
+    private ScheduledFuture<?> immediateRefreshJob;
+
+    @Override
+    public void dispose() {
+        logger.debug("Local Handler disposed.");
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (this.controller != null) {
+            this.controller.dispose();
+        }
+        deviceList = null;
+        sensorList = null;
+        super.dispose();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Telldus Local bridge handler.");
+        TelldusLocalConfiguration configuration = getConfigAs(TelldusLocalConfiguration.class);
+        this.controller = new TelldusLocalDeviceController(configuration, httpClient);
+        startAutomaticRefresh(configuration.refreshInterval);
+        refreshDeviceList();
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private synchronized void startAutomaticRefresh(long refreshInterval) {
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+        }
+        pollingJob = scheduler.scheduleWithFixedDelay(this::refreshDeviceList, 0, refreshInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    private void scheduleImmediateRefresh() {
+        // We schedule in 10 sec, to avoid multiple updates
+        logger.debug("Current remaining delay {}", pollingJob.getDelay(TimeUnit.SECONDS));
+        if (pollingJob.getDelay(TimeUnit.SECONDS) > REFRESH_DELAY) {
+            if (immediateRefreshJob == null || immediateRefreshJob.isDone()) {
+                immediateRefreshJob = scheduler.schedule(this::refreshDeviceList, REFRESH_DELAY, TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    synchronized void refreshDeviceList() {
+        try {
+            updateDevices(deviceList);
+            updateSensors(sensorList);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (TellstickException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.error("Failed to update", e);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.error("Failed to update", e);
+        }
+    }
+
+    private synchronized void updateDevices(TellstickLocalDevices previouslist) throws TellstickException {
+        TellstickLocalDevices newList = controller.callRestMethod(TelldusLocalDeviceController.HTTP_LOCAL_API_DEVICES,
+                TellstickLocalDevices.class);
+        logger.debug("Device list {}", newList.getDevices());
+        if (newList.getDevices() != null) {
+            if (previouslist == null) {
+                logger.debug("updateDevices, Creating devices.");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.

Reply:
Changed debug to trace!